### PR TITLE
workers + chillout example

### PR DESCRIPTION
**Pull request by tenisakb**

### CODE REVIEW

File: webnode/src/components/storage/index.js [hunk 1]
```diff
@@ -73,7 +74,29 @@ class Storage extends Component {
...
+      .map((value, index) => ({
+        dataMapHash: value,
+        chunkIdx: index
+      }))
+      .map(obj => workerFindTreasure(obj));
+
+    chilloutForEach([1], () => {
+      Promise.all(transformedMap); //TODO better implementation
+    }).then(() => {});
   }
```
Comment (rfornea) [inline webnode/src/components/storage/index.js]: If I'm understanding this PR correctly, basically what you're doing is duplicating a lot of the logic that we already do, except moving it into Chillout?  And once we verify it works, we'd remove the old logic so it's only the stuff that's inside Chillout?

File: webnode/package.json [hunk 2]
```diff
@@ -5,6 +5,7 @@
   "dependencies": {
```
[the remark this answers is not in the document]
Comment (tenisakb) [reply webnode/package.json]: I didnt see other webworkers - but during some time I found that epics - rxjs has fromWork - https://xgrommx.github.io/rx-book/content/rxjs_bindings/dom/web_workers/from_web_worker.html
but still think redux-middleware is better... but for something smaller its good

File: webnode/src/redux/middlewares/workers.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import createWorkerMiddleware from "redux-worker-middleware";
+
+const FindTreasureWorker = require("worker-loader!redux/workers/findTreasureWorker"); // eslint-disable-line import/no-webpack-loader-syntax
```
[the remark this answers is not in the document]
Comment (tenisakb) [reply webnode/src/redux/middlewares/workers.js]: worker-loader - webpack worker loader  https://github.com/webpack-contrib/worker-loader  - its recommended by redux-webworker-middleware

File: webnode/src/components/storage/index.js (hunk 1 above)
Comment (tenisakb) [reply webnode/src/components/storage/index.js]: chiilout is basicly cycling code and inside it is managing performance

File: webnode/src/redux/actions/treasure-hunt-actions.js [hunk 4]
```diff
@@ -22,6 +24,7 @@ const ACTIONS = Object.freeze({
   TREASURE_HUNT_FIND_TREASURE,
   TREASURE_HUNT_SAVE_TREASURE,
   TREASURE_HUNT_INCREMENT_CHUNK,
+  TREASURE_HUNT_WORKER_FIND_TREASURE,
```
Comment (EdmundMai) [inline webnode/src/redux/actions/treasure-hunt-actions.js]: i don't think you need this?

File: webnode/src/redux/actions/treasure-hunt-actions.js [hunk 5]
```diff
@@ -68,6 +71,13 @@ const ACTIONS = Object.freeze({
   claimTreasureSuccess: ({ genesisHash, sectorIdx }) => ({
     type: TREASURE_HUNT_CLAIM_TREASURE_SUCCESS,
     payload: { genesisHash, sectorIdx }
+  }),
+  workerFindTreasure: ({ dataMapHash, chunkIdx }) => ({
+    type: TREASURE_HUNT_WORKER_FIND_TREASURE,
+    meta: {
+      WebWorker: true
+    },
+    payload: { dataMapHash, chunkIdx }
```
Comment (EdmundMai) [inline webnode/src/redux/actions/treasure-hunt-actions.js]: what is this for?

File: webnode/src/redux/epics/treasure-hunt-epics.js [hunk 6]
```diff
@@ -88,15 +86,25 @@ const performPowEpic = (action$, store) => {
...
 
+const workerfindTreasureEpic = (action$, store) => {
+  return action$
+    .ofType(treasureHuntActions.TREASURE_HUNT_WORKER_FIND_TREASURE)
+    .mergeMap(action => {
+      const { dataMapHash, chunkIdx } = action.payload;
+      console.log("workerfindTreasureEpic");
+      return Observable.empty();
+    });
+};
```
Comment (EdmundMai) [inline webnode/src/redux/epics/treasure-hunt-epics.js]: this too. this does nothing

File: webnode/src/redux/middlewares/workers.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (rfornea) [reply webnode/src/redux/middlewares/workers.js]: Yeah, but right now you have this:
"worker-loader!redux/workers/findTreasureWorker"
Just making sure it wasn't supposed to be this:
"worker-loader/redux/workers/findTreasureWorker"
Or this:
"worker-loader-redux/workers/findTreasureWorker"

**Pull request closed without merging**